### DISCUSSION TITLE
New Payement gateway - choose the best options

### DISCUSSION THREAD

**Discussion by slamer59** [saleor/saleor#10422] [Q&A] [answered]
Hello,
since the refactoring of Saleor I am a bit confused on which option will be easier to develop and easier to maintain. 

The targetted payment gateway is [Paydunya](https://developers.paydunya.com/doc/EN/introduction)

They have what they called PAR (Payment with redirection) and PSR (without redirection). For simplicity they suggest with redirection (they integrate the process for us).

They have Nodejs, python client. 
For saleor, plugin (python) or app ([Nodejs in monorepo]() or [python framework](https://mirumee.github.io/saleor-app-framework-python/)). 

Plugin is integrated with saleor core when apps have a dedicated server (express or FastAPI). 

The new reactstore seems to integrate with Saleor Apps (I don't see payment plugins).

Is there any documentation for implementing a new gateway for an app ? In [Saleor Framework](https://github.com/mirumee/saleor-app-framework-python), nodejs like **ayden** ? 

Thanks for any clarification :)

**Comment by patrys** [comment, marked as answer]
I recommend that you look at implementing this as an App. We officially support and recommend Node.js as the platform (you can find [Adyen and Mollie implementations](https://github.com/saleor/react-storefront/tree/main/apps/saleor-app-checkout/backend/payments/providers) in the monorepo) but our friends at Mirumee Software also maintain a Python framework if that's your preference.

Plugins are getting slowly replaced by Apps as we aim to eventually deprecate plugins entirely.

**Reply by slamer59** [reply]
And stripe ! 3 hours ago :) 

It is still not clear about the integration but I will try :) 

Thanks for your fast response. Really valuable.

**Reply by ashishkpaul** [reply]
hello @slamer59, i am also strugling to configure stripe plugin. If you done it sucessfully, can you write the step by step guide here. Thanks in advance

**Reply by patrys** [reply]
@ashishkpaul Please start a separate discussion about your problem. This thread is already marked as answered.
